javascripts/connectors/before-list-area/tc-right-sidebar.js

```diff
@@ -8,20 +8,37 @@ export default {
 
       @discourseComputed(
         "router.currentRouteName",
+        "router.currentRoute.attributes.category",
         "router.currentRoute.attributes.category.slug",
         "router.currentRoute.attributes.tag.id"
       )
-      showSidebar(currentRouteName, categorySlug, tagId) {
+      showSidebar(currentRouteName, category, categorySlug, tagId) {
         if (this.site.mobileView) {
           return false;
         }
 
         if (settings.show_in_routes !== "") {
           const selectedRoutes = settings.show_in_routes.split("|");
+          let subcategory = null;
+          let parentCategory = null;
+
+          // check if current page is subcategory
+          // -- is category
+          // -- does not have children itself
+          // -- has a parent category
+          if (
+            !!this.category &&
+            !this.category.has_children &&
+            !!this.category.parent_category_id
+          ) {
+            subcategory = categorySlug;
+            parentCategory = category.ancestors[0].slug;
+          }
 
           if (
             selectedRoutes.includes(currentRouteName) ||
             selectedRoutes.includes(`c/${categorySlug}`) ||
+            selectedRoutes.includes(`c/${parentCategory}/${subcategory}`) ||
             selectedRoutes.includes(`tag/${tagId}`)
           ) {
```

Comment:
I might be missing something but it feels like you could do: `selectedRoutes.includes(this.category.path)` and remove a lot of logic of this codeblock.

Reply:
I'll give that a shot! No reason, I just wanted to make it as specific as possible just in case.